src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1099,17 +1100,35 @@ public async Task GoAwayFrame_AbortAllPendingStreams_StreamFailWithExpectedExcep
 
                 // Send a GOAWAY frame that indicates that we will abort all the requests.
                 var goAwayFrame = new GoAwayFrame(0, (int)ProtocolErrors.ENHANCE_YOUR_CALM, new byte[0], 0);
-                await connection.WriteFrameAsync(goAwayFrame);
+
+                Task oldRequest = Task.Run(async () =>
```

Comment:
cc: @stephentoub @geoffkizer note how only retry on request which client has not started processing.

Reply:
This seems weird. Why aren't we retrying requests 2 and 3? The server indicated that it hasn't started processing them, so these should be retried, shouldn't they?

Reply:
That's true but it happened after client has already started receiving data which it might have acted upon